src/__tests__/medications/Medications.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { act } from '@testing-library/react'
+import { mount } from 'enzyme'
+import React from 'react'
+import { Provider } from 'react-redux'
+import { MemoryRouter } from 'react-router-dom'
+import createMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
+
+import Medications from '../../medications/Medications'
+import NewMedicationRequest from '../../medications/requests/NewMedicationRequest'
+import ViewMedication from '../../medications/ViewMedication'
+import MedicationRepository from '../../shared/db/MedicationRepository'
+import PatientRepository from '../../shared/db/PatientRepository'
+import Medication from '../../shared/model/Medication'
+import Patient from '../../shared/model/Patient'
+import Permissions from '../../shared/model/Permissions'
+import { RootState } from '../../shared/store'
+
+const mockStore = createMockStore<RootState, any>([thunk])
+
+describe('Medications', () => {
+  jest.spyOn(MedicationRepository, 'findAll').mockResolvedValue([])
+  jest
+    .spyOn(MedicationRepository, 'find')
+    .mockResolvedValue({ id: '1234', requestedOn: new Date().toISOString() } as Medication)
+  jest
+    .spyOn(PatientRepository, 'find')
+    .mockResolvedValue({ id: '12345', fullName: 'test test' } as Patient)
+
+  describe('routing', () => {
+    describe('/medications/new', () => {
+      it('should render the new medication request screen when /medications/new is accessed', () => {
+        const store = mockStore({
+          title: 'test',
+          user: { permissions: [Permissions.RequestMedication] },
+          breadcrumbs: { breadcrumbs: [] },
+          components: { sidebarCollapsed: false },
+          medication: {
+            medication: ({ id: 'medicationId', patientId: 'patientId' } as unknown) as Medication,
+            patient: { id: 'patientId', fullName: 'some name' },
+            error: {},
+          },
+        } as any)
+
+        const wrapper = mount(
+          <Provider store={store}>
+            <MemoryRouter initialEntries={['/medications/new']}>
+              <Medications />
+            </MemoryRouter>
+          </Provider>,
+        )
```

Comment:
this sort of setup is duplicated a lot in this file. To remove some duplication, but to allow for tests to be independent, I think we should make a `setup` function. 

This setup function may look something like: 

```ts
const setup = () => {
        const store = mockStore({
          title: 'test',
          user: { permissions: [Permissions.RequestMedication] },
          breadcrumbs: { breadcrumbs: [] },
          components: { sidebarCollapsed: false },
          medication: {
            medication: ({ id: 'medicationId', patientId: 'patientId' } as unknown) as Medication,
            patient: { id: 'patientId', fullName: 'some name' },
            error: {},
          },
        } as any)

        const wrapper = mount(
          <Provider store={store}>
            <MemoryRouter initialEntries={['/medications/new']}>
              <Medications />
            </MemoryRouter>
          </Provider>,
        )
	return { wrapper: wrapper as ReactWrapper, store }
}

For setup values that change across file the file, we can pass those as parameters to the function.

The goal is to eliminate any "global" setup such as mocks or initialization of variables. The setup function should be a "pure" function. It should take in an arguments that are needed, setup the variables, and then return any values that will be tested.